core/server/worker/src/main/java/alluxio/worker/file/FileDataManager.java

```diff
@@ -139,32 +138,33 @@ public boolean needPersistence(long fileId) {
    */
   public boolean isFilePersisted(long fileId) {
     synchronized (mLock) {
-      return mPersistedFiles.contains(fileId);
+      return mPersistedFilesInfo.containsKey(fileId);
     }
   }
 
   /**
    * Adds a file as persisted.
    *
    * @param fileId the file id
+   * @param ufsFileStatus the ufs file status of the persisted file
    */
-  private void addPersistedFile(long fileId) {
+  private void addPersistedFile(long fileId, UfsFileStatus ufsFileStatus) {
     synchronized (mLock) {
-      mPersistedFiles.add(fileId);
+      mPersistedFilesInfo.put(fileId, ufsFileStatus);
     }
   }
 
   /**
    * Checks if the given file exists in the under storage system.
    *
    * @param fileId the file id
-   * @return true if the file exists in under storage system, false otherwise
+   * @return the file status of the ufs file if exists, null otherwise
    */
-  private synchronized boolean fileExistsInUfs(long fileId) throws IOException {
+  private synchronized UfsFileStatus fileExistsInUfs(long fileId) throws IOException {
```

Comment:
update the method name and doc now that it gets the file status from the ufs instead of just checking for existence.

Reply:
Done